src/Models/Market.php

```diff
@@ -52,6 +57,16 @@ protected static function boot()
             }
         });
 
+        static::updating(function ($market) {
+            $other_market = Market::where('slug', $market->slug)->where('id', "<>", $market->id)->first();
+
+            if ($other_market) {
+                throw new \Exception("there is a market with the slug selected");
+            }
+            
+            Page::whereId($market->page->id)->update(['slug' => $market->slug, 'title' => $market->title]);
```

Comment:
If you guys decide to retain these duplicated fields in both models, you'll need to setup an event listener in this package that listens for changes to Page models and, if the Page model is has a relation to a market or location, then the change needs to propagate to the market or location.   But again, I question the benefit of duplicating the fields in the first place vs just having them on the Page model.